tests/unit/test_application.py

```diff
@@ -62,6 +62,235 @@
     plugin: nil
 """
 
+FULL_PROJECT_YAML = """
+name: myproject
+version: 1.0
+parts:
+  mypart:
+    plugin: nil
+    source: on-amd64-to-riscv64
```

Comment:
Maybe better to name these things like `non-grammar-source` to differentiate this from what we're doing in grammar?